keyboards/gmmk/pro/keymaps/jonavin/keymap.c

```diff
@@ -117,12 +97,93 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 
 };
 
+
+// TIMEOUTS
+#define TIMEOUT_THRESHOLD_DEFAULT   5    // default timeout minutes
+#define TIMEOUT_THRESHOLD_MAX       140  // upper limits (2 hours and 10 minutes -- no rgb indicators above this value)
+static uint16_t timeout_timer = 0;
+static uint16_t timeout_counter = 0;  //in minute intervals
+static uint16_t timeout_threshold = TIMEOUT_THRESHOLD_DEFAULT;
+
+void timeout_reset_timer(void) {
+    timeout_timer = timer_read();
+    timeout_counter = 0;
+};
+
+void timeout_update_threshold(bool increase) {
+    if (increase && timeout_threshold < TIMEOUT_THRESHOLD_MAX) timeout_threshold++;
+    if (!increase && timeout_threshold > 0) timeout_threshold--;
+};
+
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+    case KC_00:
+        if (record->event.pressed) {
+            // when keycode KC_00 is pressed
+            SEND_STRING("00");
+        } else {
+            // when keycode KC_00 is released
+        }
+        break;
+    case KC_WINLCK:
+        if (record->event.pressed) {
+            _isWinKeyDisabled = !_isWinKeyDisabled; //toggle status
+            if(_isWinKeyDisabled) {
+                process_magic(GUI_OFF, record);
+            } else {
+                process_magic(GUI_ON, record);
+            }
+        } else  unregister_code16(keycode);
+        break;
+    case RGB_TOI:
+        if(record->event.pressed) {
+            timeout_update_threshold(true);
+        } else  unregister_code16(keycode);
+        break;
+    case RGB_TOD:
+        if(record->event.pressed) {
+             timeout_update_threshold(false);  //decrease timeout
+        } else  unregister_code16(keycode);
+        break;
+    default:
+        if (record->event.pressed) { //reset activity timer
+            #ifdef RGB_MATRIX_ENABLE
+                rgb_matrix_enable();
+            #endif
+            timeout_reset_timer();
+        }
+        break;
+    }
+    return true;
+};
+
+void matrix_scan_user(void) {
+    if (timeout_threshold > 0) {
```

Comment:
Is there a reason that `RGB_DISABLE_TIMEOUT` can't be used instead of this? 

It looks like it's using an unsigned 32 bit integer, so should be able to handle 5 minutes.

Reply:
RGB_DISABLE_TIMEOUT has to be defined at compile time?  This doesn't allow you to dynamically change the timeout duration like I have it setup. It can be anywhere from 0 (turned off) to 140 minutes. And changed using Fn+< and Fn+> in the keymap.